Yank/repex.py

```diff
@@ -159,21 +251,57 @@ def open(self, mode='r'):
                 ncfile.Conventions = 'ReplicaExchange'
                 ncfile.ConventionVersion = '0.2'
                 ncfile.DataUsedFor = nc_name
-                ncfile.checkpoint_interval = self._checkpoint_interval
+                ncfile.CheckpointInterval = checkpoint_interval
+
+                # Create and initilize the global variables
+                nc_last_good_iter = ncfile.createVariable('last_iteration', int, 'scalar')
+                nc_last_good_iter[0] = 0
+                return True
+            else:
+                return False
+
+        # Setup the primary file if needed
+        # Generate a unique UUID in case we need to assign it
+        # primary and subfiles are linked by UUID
+        # Uses uuid4 to avoid assigning hostname information
+        primary_uuid = str(uuid.uuid4())
```

Comment:
I can't quite do that since the checkpoint file can move around and possibly be replaced. The UUID is really tied the primary file and so it can be safely written to as is. However, this takes into account the possibility of subfiles being left behind and the primary file being moved elsewhere post-creation. At that point, they would need the uuid from the existing file and not one created at run time, so I had to split the operation up into separate loops of "check the primary file" then "check the subfiles."

For the case of the checkpoint file, it could be reduced to that, but I'm thinking in case we use other subfiles later.